app/controllers/devise_token_auth/concerns/set_user_by_token.rb

```diff
@@ -68,17 +68,17 @@ def set_user_by_token(mapping=nil)
     return @resource if @resource && @resource.is_a?(rc)
 
     # ensure we clear the client_id
-    if !@token
-      @client_id = nil
+    if !@token.token
+      @token.client_id = nil
       return
     end
 
-    return false unless @token
+    return false unless @token.token
```

Comment:
Will this line ever get evaluated? isn't `if !@token.token` the same as `unless @token.token`?

Unless I'm totally missing something, I think this line can be refactored into the conditional check above.

Reply:
You are on a right way.
I just implemented TokenFactory. You ask me about legacy code. I was trying not touch code to show that all works properly. I think that code refactoring is the next steps, but is not approached for current pr.

Reply:
Yeah, I hear you on holding off refactoring. I'm having similar challenges with the multiple auth provider feature.